src/azul/service/responseobjects/storage_service.py

```diff
@@ -135,6 +149,24 @@ def __complete(self):
                          exc_info=exception)
             self.__abort()
             raise MultipartUploadError(self.bucket_name, self.object_key) from exception
+        else:
```

Comment:
This feels out of place, as if it were tagged on. Setting the name tag on an object is orthogonal to how the object was created. MultipartUploadHandler should only handle the MP upload, not what happens with the objects afterwards.